### ISSUE TITLE
[Feature request] PIN protection

### ISSUE THREAD

**Issue by hieu-van** [walleth/walleth#284]

**Comment by ligi** [comment]
Currently working on this - but would be really good to know your attack vectors and use-case.

**Comment by leafcutterant** [comment]
If you add app-level protection, please don't do just PIN (numeric) but allow a password (alphanumeric + symbols). Plain PINs allow only very low entropy/character, and a potential character limit can ensure that the protection is not reliable. This is a major pain point for me in Electrum, let's not make the same mistake here.

**Comment by ligi** [comment]
Yes you are complete correct. I will start with pin though as the problem is for words I need to create my own keyboard (I have one for PIN through the TEZOR support) - I do not really want to invoke the external (System Keyboard) without warning the user that the password might leak to unknown parties. Ideally I should open another issue for this - but I am now at the afterparty from the hackathon (really nice btw - you should come around)

**Comment by leafcutterant** [comment]
Indeed, only an internal keyboard is reliable and that's more work to implement. [Electrum](https://github.com/spesmilo/electrum) for Android does have one, maybe you can use some of their code.

I had to skip the afterparty due to work, but hopefully see you at the Status hackathon and other events!
